extmod/ntptime.py

```diff
@@ -26,14 +25,26 @@ def time():
     finally:
         s.close()
     val = struct.unpack("!I", msg[40:44])[0]
+
+    EPOCH_YEAR = time.gmtime(0)[0]
+    if EPOCH_YEAR == 2000:
+        # (date(2000, 1, 1) - date(1900, 1, 1)).days * 24*60*60
+        NTP_DELTA = 3155673600
+    elif EPOCH_YEAR == 1970:
+        # (date(1970, 1, 1) - date(1900, 1, 1)).days * 24*60*60
+        NTP_DELTA = 2208988800
+    else:
+        raise Exception(
+            f"Unsupported EPOCH_YEAR provided in this port of MicroPython: {EPOCH_YEAR}. Cannot interpret NTP time."
```

Comment:
I think it's better not to use f-strings, they are a relatively new feature to uPy and not enabled on all ports/boards.  Also this error message could be made a bit shorter, eg `"unsupported Epoch {}".format(EPOCH_YEAR)`

Reply:
No problem - done!